spanner/leaderboard/complete/src/main/java/com/google/codelabs/App.java

```diff
@@ -0,0 +1,353 @@
+    String score;
+    Statement statement =
+        Statement
+            .newBuilder(
+              "SELECT p.PlayerId, p.PlayerName, s.Score, s.Timestamp "
+              + "FROM Players p "
+              + "JOIN Scores s ON p.PlayerId = s.PlayerId "
+              + "WHERE s.Timestamp > "
+              + "TIMESTAMP_SUB(CURRENT_TIMESTAMP(), "
+              + "    INTERVAL @Timespan HOUR) "
+              + "ORDER BY s.Score DESC LIMIT 10")
+            .bind("Timespan")
+            .to(timespan)
+            .build();
+    ResultSet resultSet =
+        dbClient
+            .singleUse()
+            .executeQuery(statement);
+    while (resultSet.next()) {
+      scoreDate = String.valueOf(resultSet.getTimestamp("Timestamp"));
+      score = String.format("%,d", resultSet.getLong("Score"));
+      System.out.printf(
+          "PlayerId: %d  PlayerName: %s  Score: %s  Timestamp: %s\n",
+          resultSet.getLong("PlayerId"), resultSet.getString("PlayerName"), score,
+          scoreDate.substring(0,10));
+    }
+  }
+
+  static void delete(DatabaseAdminClient dbAdminClient, DatabaseId db) {
+    try  {
+      dbAdminClient.dropDatabase(db.getInstanceId().getInstance(), db.getDatabase());
+    } catch (Exception e) {
+      System.err.println("Error encountered while deleting database.");
+    }
+    System.out.printf("Database deleted.\n");
+  }
+
+  static void printUsageAndExit() {
+    System.out.println("Leaderboard 1.0.0");
+    System.out.println("Usage:");
+    System.out.println("  java -jar leaderboard.jar "
+        + "<command> <instance_id> <database_id> [command_option]");
+    System.out.println("");
+    System.out.println("Examples:");
+    System.out.println("  java -jar leaderboard.jar create my-instance example-db");
+    System.out.println("      - Create a sample Cloud Spanner database along with "
+        + "sample tables in your project.\n");
+    System.out.println("  java -jar leaderboard.jar insert my-instance example-db players");
+    System.out.println("      - Insert 100 sample Player records into the database.\n");
+    System.out.println("  java -jar leaderboard.jar insert my-instance example-db scores");
+    System.out.println("      - Insert sample score data into Scores sample Cloud Spanner "
+        + "database table.\n");
+    System.out.println("  java -jar leaderboard.jar query my-instance example-db");
+    System.out.println("      - Query players with top ten scores of all time.\n");
+    System.out.println("  java -jar leaderboard.jar query my-instance example-db 168");
+    System.out.println("      - Query players with top ten scores within a timespan "
+        + "specified in hours.\n");
+    System.out.println("  java -jar leaderboard.jar delete my-instance example-db");
+    System.out.println("      - Delete sample Cloud Spanner database.");
+    System.exit(1);
+  }
+
+  public static void main(String[] args) throws Exception {
+    if (!(args.length == 3 || args.length == 4)) {
```

Comment:
For CLI stuff I would recommend [ArgParse](https://argparse4j.github.io/). It can really help reduce some of the boiler plate stuff and will handle error messages and incorrect args much more throughly. It's up to you if you want to implement it here since you've already done it all, but it might be catch some edge cases.

Reply:
Opting to leave as is to minimize dependencies for getting started.